eslint.config.mjs

```diff
@@ -28,7 +42,18 @@ export default tseslint.config(
     },
     languageOptions: {
       parserOptions: {
-        projectService: true,
+        projectService: {
+          // We must enumerate each file that we want to lint that is not explicitly captured by a project-specific tsconfig
+          // allowDefaultProject does not allow `**` globs.
+          // https://github.com/typescript-eslint/typescript-eslint/issues/9739
+          allowDefaultProject: allProjectDirs.map(dir => `${dir}/*.ts`)
+        },
```

Comment:
This fixes the issue -- any top-level `.ts` file will be linted against the top-level tsconfig.json